diff --git a/config.h b/config.h
index 93a875a..e9f572b 100644
--- a/config.def.h
+++ b/config.def.h
@@ -65,5 +65,20 @@ static const char unknown_str[] = "n/a";
  */
 static const struct arg args[] = {
 	/* function format          argument */
-	{ datetime, "%s",           "%F %T" },
+        { run_command, "  %s  |", "tsm-status" },
+        { wifi_essid, "  󰖩  %s  |  ",  "wlan0" },
+        { netspeed_rx, "󰅀  %s |  ",  "wlan0" }, // receive network speed
+        { netspeed_tx, "󰅃  %s |  ",  "wlan0" }, // transfer network speed
+        { cpu_perc, "  %s%%  |  ",  NULL },
+        { ram_perc, "󰀹  %s%%  |  ",  NULL },
+        { battery_perc, "  %s%%",   "BAT0" },
+        // { battery_state, " %s ", "BAT0" },
+        { battery_remaining, " %s  |  ", "BAT0" },
+        { run_command, "󰂯  %s  |  ", "bluetoothctl devices Connected | awk '{print$3}'" },
+        { run_command, "󰕾  %s  |  ", "pamixer --get-volume-human" },
+        // { vol_perc, "  %s  |  ", "Master" }, # https://tools.suckless.org/slstatus/patches/alsa/slstatus-alsa-4bd78c9.patch
+        { backlight_perc, "󰃠  %s  |  ", "intel_backlight" },
+        // { uptime, "  %s  |  ", NULL },
+        // { entropy, "  %s  |  ", NULL },
+        { datetime, "%s  ", "%a %b %d %R" },
 };
