diff --git a/dmenu.c b/dmenu.c
index 0361929..74d67d5 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -449,7 +449,7 @@ keypress(XKeyEvent *ev)
 		case XK_e: ksym = XK_End;       break;
 		case XK_f: ksym = XK_Right;     break;
 		case XK_g: ksym = XK_Escape;    break;
-		case XK_h: ksym = XK_BackSpace; break;
+		case XK_o: ksym = XK_BackSpace; break;
 		case XK_i: ksym = XK_Tab;       break;
 		case XK_j: /* fallthrough */
 		case XK_J: /* fallthrough */
@@ -457,6 +457,8 @@ keypress(XKeyEvent *ev)
 		case XK_M: ksym = XK_Return; ev->state &= ~ControlMask; break;
 		case XK_n: ksym = XK_Down;      break;
 		case XK_p: ksym = XK_Up;        break;
+		case XK_l: ksym = XK_Down;      break;
+		case XK_h: ksym = XK_Up;        break;
 
 		case XK_k: /* delete right */
 			text[cursor] = '\0';
